nikola/plugins/command/new_post.py

```diff
@@ -265,16 +265,33 @@ def _execute(self, options, args):
         if "@" in content_format:
             content_format, content_subformat = content_format.split("@")
 
-        if not content_format:  # Issue #400
+        if not content_format and path:
+            # content_format not specified. If path was given, use
+            # it to guess (Issue #2798)
+            extension = os.path.splitext(path)[-1]
+            for compiler, extensions in self.site.config['COMPILERS'].items():
+                if extension in extensions:
+                    content_format = compiler
+
+        elif not content_format and import_file:
```

Comment:
Perhaps, but it saves 3 lines of code at the cost of a non-stupid condition. I like stupid conditions.